sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -4,17 +4,29 @@
 # --------------------------------------------------------------------------------------------
 import logging
 import threading
-
-from typing import Any, Union, TYPE_CHECKING, Dict, List, Optional, cast
+from typing import (
+    Any,
+    Union,
+    TYPE_CHECKING,
+    Dict,
+    List,
+    Optional,
+    Callable,
+    cast,
+    overload,
+)
+from typing_extensions import Literal
```

Comment:
I think Johan/Laurent mentioned that if we want to import from typing_extensions from now on, we need to include it in the required installs here: https://github.com/Azure/azure-sdk-for-python/blob/bc31f1674fd6e20d5c6eb00ab7c012896afd5ca8/sdk/eventhub/azure-eventhub/setup.py#L73

I believe they said that it's a dependency for azure-core right now, but might not be forever.